tests/unit_tests/seraphis_cache.cpp

```diff
@@ -0,0 +1,74 @@
+#include "seraphis_cache.h"
+
+#include <boost/range/adaptor/reversed.hpp>
+#include <gtest/gtest.h>
+
+bool HalfPruning::should_prune(const std::deque<uint64_t> &window, uint64_t chain_top)
+{
+    if (chain_top - *window.end() <= m_num_unprunable)
```

Comment:
I think this will cause the unprunable region to be larger than your config specifies, since if the middle of the range is pruned then the region between the middle and the end will never be pruned in the case where this condition is right on the edge.

Reply:
the size when should_prune is called is constant. the other stuff is my stupidity and lack of time spent on this. currently focused on making this work.
for some reason I had the impression that the window is ordered forwards, no idea why. for the other thing, yea, thats why that weird resize (I added that yesterday without any thought put into it.)
will have more brain power over this tomorrow

Reply:
ah, I just forgot I was (supposed to be) going from the top. sorry for all these, not opening my journey in monero great.